gateway/ln-gateway/src/lnd.rs

```diff
@@ -263,10 +264,71 @@ impl ILnRpcClient for GatewayLndClient {
     }
 
     async fn routehints(&self) -> crate::Result<GetRouteHintsResponse> {
-        // TODO: Issue #1953: Implement full route hint fetching for LND gateways
-        Ok(GetRouteHintsResponse {
-            route_hints: vec![RouteHint { hops: vec![] }],
-        })
+        let mut client = self.client.clone();
+        let channels = client
+            .lightning()
+            .list_channels(ListChannelsRequest {
+                active_only: true,
+                inactive_only: false,
+                public_only: false,
+                private_only: false,
+                peer: vec![],
+            })
+            .await
+            .map_err(|e| {
+                GatewayError::LnRpcError(tonic::Status::new(
+                    tonic::Code::Internal,
+                    format!("LND error: {e:?}"),
+                ))
+            })?
+            .into_inner();
+
+        let mut route_hints: Vec<RouteHint> = vec![];
+        for chan in channels.channels {
+            let info = client
+                .lightning()
+                .get_chan_info(ChanInfoRequest {
+                    chan_id: chan.chan_id,
+                })
+                .await
+                .map_err(|e| {
+                    GatewayError::LnRpcError(tonic::Status::new(
+                        tonic::Code::Internal,
+                        format!("LND error: {e:?}"),
+                    ))
+                })?
+                .into_inner();
+
+            let policy = match info.node1_policy.clone() {
+                Some(policy) => policy,
+                None => continue,
+            };
+            let src_node_id = PublicKey::from_str(&chan.remote_pubkey)
+                .unwrap()
+                .serialize()
+                .to_vec();
```

Comment:
Is there a better way to do this?

Reply:
The way you do it *might* be the right one because it validates that we got a valid pub key. If you don't want that it looks like simply parsing a hex string?